Index: linux-3.18.84/arch/mips/ath79/Kconfig
===================================================================
--- linux-3.18.84.orig/arch/mips/ath79/Kconfig  2022-07-02 23:41:21.269860000 -0300
+++ linux-3.18.84/arch/mips/ath79/Kconfig       2022-07-02 22:45:24.000000000 -0300
@@ -480,6 +480,22 @@
        select ATH79_DEV_NFC
        select ATH79_DEV_USB
 
+config ATH79_MACH_DOMYWIFI_DW33D
+       bool "DomyWifi DW33D support"
+       select SOC_QCA955X
+       select ATH79_DEV_AP9X_PCI if PCI
+       select ATH79_DEV_ETH
+       select ATH79_DEV_GPIO_BUTTONS
+       select ATH79_DEV_LEDS_GPIO
+       select ATH79_DEV_M25P80
+       select ATH79_DEV_NFC
+       select ATH79_DEV_WMAC
+       select ATH79_DEV_USB
+       select MTD_NAND
+       select MTD_NAND_AR934X
+       select MTD_NAND_AR934X_HW_ECC
+       select MTD_NAND_ECC
+
 config ATH79_MACH_DRAGINO2
        bool "DRAGINO V2 support"
        select SOC_AR933X
@@ -1180,6 +1196,15 @@
        select ATH79_DEV_M25P80
        select ATH79_DEV_WMAC
 
+config ATH79_MACH_TL_WR941HP_V2
+       bool "TP-LINK TL-WR941HP v2 support"
+       select SOC_QCA956X
+       select ATH79_DEV_ETH
+       select ATH79_DEV_GPIO_BUTTONS
+       select ATH79_DEV_LEDS_GPIO
+       select ATH79_DEV_M25P80
+       select ATH79_DEV_WMAC
+
 config ATH79_MACH_TL_WR1041N_V2
        bool "TP-LINK TL-WR1041N v2 support"
        select SOC_AR934X
@@ -1476,4 +1501,4 @@
 config PCI_AR724X
        def_bool n
 
-endif
+endif
\ No newline at end of file
Index: linux-3.18.84/arch/mips/ath79/Makefile
===================================================================
--- linux-3.18.84.orig/arch/mips/ath79/Makefile 2022-07-02 23:41:21.269860000 -0300
+++ linux-3.18.84/arch/mips/ath79/Makefile      2022-07-02 22:48:27.000000000 -0300
@@ -69,6 +69,7 @@
 obj-$(CONFIG_ATH79_MACH_DIR_615_I1)    += mach-dir-615-i1.o
 obj-$(CONFIG_ATH79_MACH_DIR_825_B1)    += mach-dir-825-b1.o
 obj-$(CONFIG_ATH79_MACH_DIR_825_C1)    += mach-dir-825-c1.o
+obj-$(CONFIG_ATH79_MACH_DOMYWIFI_DW33D)        += mach-domywifi-dw33d.o
 obj-$(CONFIG_ATH79_MACH_DRAGINO2)      += mach-dragino2.o
 obj-$(CONFIG_ATH79_MACH_ESR900)        += mach-esr900.o
 obj-$(CONFIG_ATH79_MACH_EW_DORIN)      += mach-ew-dorin.o
@@ -141,6 +142,7 @@
 obj-$(CONFIG_ATH79_MACH_TL_WR841N_V9)  += mach-tl-wr841n-v9.o
 obj-$(CONFIG_ATH79_MACH_TL_WR941ND)    += mach-tl-wr941nd.o
 obj-$(CONFIG_ATH79_MACH_TL_WR941ND_V6) += mach-tl-wr941nd-v6.o
+obj-$(CONFIG_ATH79_MACH_TL_WR941HP_V2) += mach-tl-wr941hp-v2.o
 obj-$(CONFIG_ATH79_MACH_TL_WR1041N_V2) += mach-tl-wr1041n-v2.o
 obj-$(CONFIG_ATH79_MACH_TL_WR1043ND)   += mach-tl-wr1043nd.o
 obj-$(CONFIG_ATH79_MACH_TL_WR1043ND_V2)        += mach-tl-wr1043nd-v2.o
 @@ -176,4 +178,4 @@
 obj-$(CONFIG_ATH79_MACH_WZR_450HP2)    += mach-wzr-450hp2.o
 obj-$(CONFIG_ATH79_MACH_ZCN_1523H)     += mach-zcn-1523h.o
 obj-$(CONFIG_ATH79_MACH_CARAMBOLA2)    += mach-carambola2.o
-obj-$(CONFIG_ATH79_MACH_NBG6716)       += mach-nbg6716.o
+obj-$(CONFIG_ATH79_MACH_NBG6716)       += mach-nbg6716.o
\ No newline at end of file
Index: linux-3.18.84/arch/mips/ath79/machtypes.h
===================================================================
--- linux-3.18.84.orig/arch/mips/ath79/machtypes.h      2022-07-02 23:41:21.269860000 -0300
+++ linux-3.18.84/arch/mips/ath79/machtypes.h   2022-07-02 22:43:17.000000000 -0300
@@ -57,6 +57,7 @@
        ATH79_MACH_DIR_835_A1,          /* D-Link DIR-835 rev. A1 */
        ATH79_MACH_DLAN_PRO_500_WP,     /* devolo dLAN pro 500 Wireless+ */
        ATH79_MACH_DLAN_PRO_1200_AC,    /* devolo dLAN pro 1200+ WiFi ac*/
+       ATH79_MACH_DOMYWIFI_DW33D,      /* DomyWifi DW33D */
        ATH79_MACH_DRAGINO2,            /* Dragino Version 2 */
        ATH79_MACH_ESR900,              /* EnGenius ESR900 */
        ATH79_MACH_EW_DORIN,            /* embedded wireless Dorin Platform */
@@ -148,7 +149,7 @@
        ATH79_MACH_TL_MR3420_V2,        /* TP-LINK TL-MR3420 v2 */
        ATH79_MACH_TL_WA701ND_V2,       /* TP-LINK TL-WA701ND v2 */
        ATH79_MACH_TL_WA750RE,          /* TP-LINK TL-WA750RE */
-       ATH79_MACH_TL_WA7210N_V2,       /* TP-LINK TL-WA7210N v2 */
+       ATH79_MACH_TL_WA7210N_V2,       /* TP-LINK TL-WA7210N v2 */
        ATH79_MACH_TL_WA7510N_V1,       /* TP-LINK TL-WA7510N v1*/
        ATH79_MACH_TL_WA850RE,          /* TP-LINK TL-WA850RE */
        ATH79_MACH_TL_WA860RE,          /* TP-LINK TL-WA860RE */
@@ -179,6 +180,7 @@
        ATH79_MACH_TL_WR941ND,          /* TP-LINK TL-WR941ND */
        ATH79_MACH_TL_WR941ND_V5,       /* TP-LINK TL-WR941ND v5 */
        ATH79_MACH_TL_WR941ND_V6,       /* TP-LINK TL-WR941ND v6 */
+       ATH79_MACH_TL_WR941HP_V2,       /* TP-LINK TL-WR941HP v2 */
        ATH79_MACH_TUBE2H,              /* Alfa Network Tube2H */
        ATH79_MACH_UBNT_AIRGW,          /* Ubiquiti AirGateway */
        ATH79_MACH_UBNT_AIRROUTER,      /* Ubiquiti AirRouter */
@@ -197,7 +199,7 @@
        ATH79_MACH_UBNT_UNIFI_OUTDOOR,  /* Ubiquiti UnifiAP Outdoor */
                ATH79_MACH_UBNT_UNIFI_OUTDOOR_PLUS, /* Ubiquiti UnifiAP Outdoor+ */
        ATH79_MACH_UBNT_XM,             /* Ubiquiti Networks XM board rev 1.0 */
-       ATH79_MACH_WEIO,                        /* WeIO board */
+       ATH79_MACH_WEIO,                /* WeIO board */
        ATH79_MACH_WHR_G301N,           /* Buffalo WHR-G301N */
        ATH79_MACH_WHR_HP_G300N,        /* Buffalo WHR-HP-G300N */
        ATH79_MACH_WHR_HP_GN,           /* Buffalo WHR-HP-GN */
@@ -231,4 +233,4 @@
        ATH79_MACH_ZCN_1523H_5,         /* Zcomax ZCN-1523H-5-xx */
 };
 
-#endif /* _ATH79_MACHTYPE_H */
+#endif /* _ATH79_MACHTYPE_H */
\ No newline at end of file